tools/data-handler/src/validate.ts

```diff
@@ -237,6 +261,58 @@ export class Validate {
         return response;
     }
 
+    /**
+     * Checks if card's current workflow state matches workflow that card's cardtype is using.
+     * @param {Project} project Project object.
+     * @param {card} card Card object to validate
+     * @returns requestStatus 200 when validation succeeded
+     *  <br> statusCode 400 when schema validation failed
+     *  <br> statusCode 500 when unknown error occurred.
+     *
+     */
+    public async validateWorkflowState(project: Project, card: card): Promise<requestStatus> {
+
+        if (!card.metadata) {
+            return {
+                statusCode: 500,
+                message: `Card '${card.key}' has no metadata. Card object needs to be instantiated with '{metadata: true}'`
+            };
+        }
+
+        const cardType = await project.cardType(card.metadata?.cardtype);
+        if (!cardType) {
+            return {
+                statusCode: 500,
+                message: `Card '${card.key}' has invalid cardtype '${card.metadata?.cardtype}'`
```

Comment:
Do you think it would be beneficial to use `errorFunction` consistantly?

Reply:
It would.